docs/dimensions.md

```diff
@@ -0,0 +1,83 @@
+
+```php
+public function onNewVisit(Request $request, Visitor $visitor, $action)
+{
+	return Common::getRequestVar('sport_activity_type', $default = false, 'string', $request->getParams());
+}
+```
+
+Whenever you send a `sport_activity_type` parameter with any tracking request and the tracker detected a new visitor, this information will be recorded in the database and can be used to create new reports or to extend existing reports and API methods. An example tracking URL could look like this `piwik.php?idsite=1&sport_activity_type=running`
+
+Optionally, you can overwrite any initial value that was written when a new visit was detected by implementing the method `onExistingVisit`. You can use this for example for counters, to store the time of the last action etc:
+
+```php
+public function onNewVisit(Request $request, Visitor $visitor, $action)
+{
+    return 1;
+}
+
+public function onExistingVisit(Request $request, Visitor $visitor, $action)
+{
+    return $visitor->getVisitorColumn($this->column_name) + 1;
+}
+```
+
+When a new visitor is being tracked, the value `1` will be stored. On each further tracking request the value for this dimension will be increased.
+
+
+## Action dimensions
+
+Action dimensions work similar, just the method name is differently. Let's say you want to track the current speed of a runner by exposing a new URL parameter `speed`:
+
+```php
+public function onNewAction(Request $request, Visitor $visitor, Action $action)
+{
+    $value = Common::getRequestVar('speed', false, 'string', $request->getParams());
+
+    $isRunner = 'running' === $visitor->getVisitorColumn('sport_activity_type');
+
+    if ($isRunner && is_numeric($value)) {
+        // only store the value if it is a runner and if speed is numeric
+        return $value;
+    }
+
+    return false; // do not store any value for this action
+}
+```
+
+A tracking request could be done like this: `piwik.php?idsite=1&sport_activity_type=running&speed=50`.
+
+Of course you can add any custom behaviour like limiting the max speed etc.
+
+
+## Segmentation
+A new segment is automatically created for a dimension and defined in the method `configureSegments`. All you usually have to do is to adjust the description of the accepted values.
+
+## Learn more
+Dimensions are quite powerful. For example you can change the behavior of an existing dimension by creating a dimension that has the same column name, you can store action related data very space efficient by using a lookup table and they can force the creation of a new visit in case an existing visitor was recognized. We recommend to have a look at the documentation within a created dimension and at the API-Reference of the classes [VisitDimension](/api-reference/Piwik/Plugin/Dimension/VisitDimension), [ActionDimension](/api-reference/Piwik/Plugin/Dimension/ActionDimension) and [ConversionDimension](/api-reference/Piwik/Plugin/Dimension/ConversionDimension)
```

Comment:
'very space efficient' => efficiently
'and they can force' => '. Dimensions can also force'